Godeps/_workspace/src/github.com/docker/libkv/store/boltdb/boltdb.go

```diff
@@ -381,7 +379,7 @@ func (b *BoltDB) AtomicPut(key string, value []byte, previous *store.KVPair, opt
 		// doesn't exist in the DB.
 		val = bucket.Get([]byte(key))
 		if previous == nil && len(val) != 0 {
-			return store.ErrKeyModified
+			return store.ErrKeyExists
```

Comment:
Yeah I think this change was made long time back in libkv. But we need to treat both `KeyExists` and `KeyModified` as key modified in datastore. Let me make that change as part of this PR lest we forget.